source/contributed/typescript-and-screeps-1.md

```diff
@@ -0,0 +1,207 @@
+title: TypeScript and Screeps, from Beginner to Advanced, Pt. 1
+contributed:
+    name: bryan
+    link: https://github.com/bryanbecker
+    date: 2017-05-15
+---
+
+# TypeScript and Screeps, from Beginner to Advanced, Pt. 1
+
+* [Part 1](http://docs.screeps.com/contributed/typescript-and-screeps-1.html)
+**Introduction to TypeScript and setting up a TypeScript + Screeps environment**
+* [Part 2](http://docs.screeps.com/contributed/typescript-and-screeps-2.html)
+In depth look using TypeScript's features to extend the built-in Screeps objects and methods
+* [Part 3](http://docs.screeps.com/contributed/typescript-and-screeps-3.html)
+Getters, setters, and meta programming with decorators
+
+Part One of this series with introduce setting up a TypeScript project for Screeps.  This article will present a basic guide to setting up a TypeScript project for Screeps, and a brief introduction to third party tools.
+
+For users who already have a working TypeScript setup, you can directly skip to [Part 2](http://docs.screeps.com/contributed/typescript-and-screeps-2.html).
+
+For a quick-start, skip down to [screeps-typescript-starter](#screeps-typescript-starter).
+
+## Why TypeScript?
+
+TypeScript is a superset of ECMAScript (a fancy way to say Javascript).  That means it has *all* the features of ECMA, *plus* some extra ones.  When a project is compiled, Typescript will process all your files, and "transpile" them into the target Javascript version (ES6, in the case of Screeps).  This works similarly to Babel, if you have used that in the past.
```

Comment:
I was trying to be cheeky... I guess it ended up being confusing... will clarify